javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -253,28 +303,39 @@ object Character {
   @inline def isSurrogatePair(high: scala.Char, low: scala.Char): scala.Boolean =
     isHighSurrogate(high) && isLowSurrogate(low)
 
-  @inline def toCodePoint(high: scala.Char, low: scala.Char): scala.Int =
+  @inline def toCodePoint(high: scala.Char, low: scala.Char): Int =
     ((high & SurrogateUsefulPartMask) << 10) + (low & SurrogateUsefulPartMask) + 0x10000
 
   // --- End of UTF-16 surrogate pairs handling ---
 
-  def isUnicodeIdentifierStart(c: scala.Char): scala.Boolean =
-    reUnicodeIdentStart.test(c.toString)
-
-  def isUnicodeIdentifierPart(c: scala.Char): scala.Boolean =
-    isUnicodeIdentifierStart(c) || isIdentifierIgnorable(c) ||
-    reUnicodeIdentPartExcl.test(c.toString)
-
-  def isIdentifierIgnorable(c: scala.Char): scala.Boolean =
-    reIdentIgnorable.test(c.toString)
+  def isLowerCase(c: scala.Char): scala.Boolean =
+    isLowerCase(c.toInt)
 
-  //def isMirrored(c: scala.Char): scala.Boolean
+  def isLowerCase(c: Int): scala.Boolean = {
+    if (c < 256)
+      '\u00AA' == c || '\u00BA' == c || getTypeLT256(c) == LOWERCASE_LETTER
+    else
+      isLowerCaseGE256(c)
+  }
 
-  def isLowerCase(c: scala.Char): scala.Boolean =
-    reIsLower.test(c.toString)
+  private[this] def isLowerCaseGE256(c: Int): scala.Boolean = {
+    ('\u02B0' <= c && c <= '\u02B8') || ('\u02C0' <= c && c <= '\u02C1') ||
+      ('\u02E0' <= c && c <= '\u02E4') || '\u0345' == c || '\u037A' == c ||
+      ('\u1D2C' <= c && c <= '\u1D6A') || '\u1D78' == c ||
+      ('\u1D9B' <= c && c <= '\u1DBF') || '\u2071' == c || '\u207F' == c ||
+      ('\u2090' <= c && c <= '\u209C') || ('\u2170' <= c && c <= '\u217F') ||
+      ('\u24D0' <= c && c <= '\u24E9') || ('\u2C7C' <= c && c <= '\u2C7D') ||
+      '\uA770' == c || ('\uA7F8' <= c && c <= '\uA7F9') ||
+      getTypeGE256(c) == LOWERCASE_LETTER
```

Comment:
I really meant *no* indentation wrt to the first line, so:
```scala
    ('\u02B0' <= c && c <= '\u02B8') || ('\u02C0' <= c && c <= '\u02C1') ||
    ('\u02E0' <= c && c <= '\u02E4') || '\u0345' == c || '\u037A' == c ||
    ...
    getTypeGE256(c) == LOWERCASE_LETTER
```
See https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#long-expressions-with-binary-operators
Same everywhere else with this pattern, of course.